OpenRA.Mods.Common/Traits/Tooltip.cs

```diff
@@ -54,18 +54,20 @@ public class TooltipInfo : TooltipInfoBase, ITooltipInfo
 		public string TooltipForPlayerStance(PlayerRelationship stance)
 		{
 			if (stance == PlayerRelationship.None || !GenericVisibility.HasStance(stance))
-				return Name;
+				return Game.Translate(Name);
+
+			var genericName = Game.Translate(GenericName);
 
 			if (GenericStancePrefix && !string.IsNullOrEmpty(AllyPrefix) && stance == PlayerRelationship.Ally)
-				return AllyPrefix + " " + GenericName;
+				return AllyPrefix + " " + genericName;
```

Comment:
Places where string concatenation is used to piece together phrases and sentences should be changed to use the full phrase as a translation. In this case for example the prefix idea might not work in many languages because of grammar. The prefix may need to agree with the noun's gender and number. So this is a good case where gender might come in play - in Bulgarian for example "building" is feminine and "tank" is masculine. So we need to have the whole string for translation - "Allied {$genericName}", "Enemy {$genericName}" and then add gender attributes to the generic name translations.

Reply:
It seems Fluent is not capable of inserting strings that are going to be translated themselves and check their gender. https://www.projectfluent.org/fluent/guide/terms.html It only allows combining static elements.